src/ResourceManager/KeyVault/Commands.KeyVault/Commands/GetAzureKeyVaultCertificateOperation.cs

```diff
@@ -41,19 +50,39 @@ public class GetAzureKeyVaultCertificateOperation : KeyVaultCmdletBase
         /// Name
         /// </summary>       
         [Parameter(Mandatory = true,
+                   ParameterSetName = ByNameParameterSet,
                    Position = 1,
                    ValueFromPipelineByPropertyName = true,
                    HelpMessage = "Certificate name. Cmdlet constructs the FQDN of a certificate operation from vault name, currently selected environment and certificate name.")]
         [ValidateNotNullOrEmpty]
         [Alias(Constants.CertificateName)]
         public string Name { get; set; }
 
+        /// <summary>
+        /// Certificate Object
+        /// </summary>
+        [Parameter(Mandatory = true,
+                   ParameterSetName = ByInputObjectParameterSet,
+                   Position = 0,
+                   ValueFromPipeline = true,
+                   HelpMessage = "Certificate Object.")]
+        [ValidateNotNullOrEmpty]
+        public PSKeyVaultCertificateIdentityItem InputObject { get; set; }
+
         #endregion
 
-        protected override void ProcessRecord()
+        public override void ExecuteCmdlet()
         {
+            if (InputObject != null)
+            {
+                VaultName = InputObject.VaultName.ToString();
+                Name = InputObject.Name.ToString();
+            }
+
             var certificateOperation = this.DataServiceClient.GetCertificateOperation(VaultName, Name);
-            var kvCertificateOperation = KeyVaultCertificateOperation.FromCertificateOperation(certificateOperation);
+            var kvCertificateOperation = PSKeyVaultCertificateOperation.FromCertificateOperation(certificateOperation);
+            kvCertificateOperation.VaultName = VaultName;
+            kvCertificateOperation.Name = Name;
```

Comment:
Are these assignments necessary here? Could they be moved into the FromCertOp method? (haven't checked if the vault name is available there)